clap-v3-utils/src/input_parsers/mod.rs

```diff
@@ -0,0 +1,477 @@
+        })
+}
+
+pub fn parse_url_or_moniker(arg: &str) -> Result<String, String> {
+    parse_url(&normalize_to_url_if_moniker(arg))
+}
+
+pub fn parse_pow2(arg: &str) -> Result<usize, String> {
+    arg.parse::<usize>()
+        .map_err(|e| format!("Unable to parse, provided: {arg}, err: {e}"))
+        .and_then(|v| {
+            v.is_power_of_two()
+                .then_some(v)
+                .ok_or(format!("Must be a power of 2: {v}"))
+        })
+}
+
+pub fn parse_percentage(arg: &str) -> Result<u8, String> {
+    arg.parse::<u8>()
+        .map_err(|e| format!("Unable to parse input percentage, provided: {arg}, err: {e}"))
+        .and_then(|v| {
+            (v <= 100).then_some(v).ok_or(format!(
+                "Percentage must be in range of 0 to 100, provided: {v}"
+            ))
+        })
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum UiTokenAmount {
+    Amount(f64),
+    All,
+}
+impl UiTokenAmount {
+    pub fn parse_amount(arg: &str) -> Result<UiTokenAmount, String> {
+        arg.parse::<f64>()
+            .map(UiTokenAmount::Amount)
+            .map_err(|_| format!("Unable to parse input amount, provided: {arg}"))
+    }
+
+    pub fn parse_amount_or_all(arg: &str) -> Result<UiTokenAmount, String> {
+        if arg == "ALL" {
+            Ok(UiTokenAmount::All)
+        } else {
+            Self::parse_amount(arg).map_err(|_| {
+                format!("Unable to parse input amount as float or 'ALL' keyword, provided: {arg}")
+            })
+        }
+    }
+
+    pub fn to_raw_amount(&self, decimals: u8) -> RawTokenAmount {
+        match self {
+            UiTokenAmount::Amount(amount) => {
+                RawTokenAmount::Amount((amount * 10_usize.pow(decimals as u32) as f64) as u64)
+            }
+            UiTokenAmount::All => RawTokenAmount::All,
+        }
+    }
+
+    pub fn sol_to_lamport(&self) -> RawTokenAmount {
+        const NATIVE_SOL_DECIMALS: u8 = 9;
+        self.to_raw_amount(NATIVE_SOL_DECIMALS)
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum RawTokenAmount {
+    Amount(u64),
+    All,
+}
+
+pub fn parse_rfc3339_datetime(arg: &str) -> Result<String, String> {
+    DateTime::parse_from_rfc3339(arg)
+        .map(|_| arg.to_string())
+        .map_err(|e| format!("{e}"))
+}
+
+pub fn parse_derivation(arg: &str) -> Result<String, String> {
+    let value = arg.replace('\'', "");
+    let mut parts = value.split('/');
+    let account = parts.next().unwrap();
+    account
+        .parse::<u32>()
+        .map_err(|e| format!("Unable to parse derivation, provided: {account}, err: {e}"))
+        .and_then(|_| {
+            if let Some(change) = parts.next() {
+                change.parse::<u32>().map_err(|e| {
+                    format!("Unable to parse derivation, provided: {change}, err: {e}")
+                })
+            } else {
+                Ok(0)
+            }
+        })?;
+    Ok(arg.to_string())
```

Comment:
Yeah, the parsers that return a `String` type basically run the original validator and then return the argument as string. It would be nice to just invoke the validator directly and return the argument, but this will trigger a deprecation warning...

Reply:
I was thinking about moving the shared logic to a helper function that's called from the validator and the parser. But it may not be worth the effort.